PLNPluginSchemaMigration.inc.php

```diff
@@ -48,14 +41,42 @@ public function up() {
 				$table->bigInteger('journal_id');
 				$table->string('uuid', 36)->nullable();
 				$table->bigInteger('status')->default(0)->nullable();
+				$table->string('staging_state')->nullable();
+				$table->string('lockss_state')->nullable();
 				$table->datetime('date_status')->nullable();
 				$table->datetime('date_created');
 				$table->datetime('date_modified')->nullable();
 				$table->string('export_deposit_error', 1000)->nullable();
+				$table->datetime('date_preserved')->nullable();
 			});
 		}
 
 		// Create a new scheduled_tasks entry for this plugin
 		Capsule::table('scheduled_tasks')->insertOrIgnore(['class_name' => 'plugins.generic.pln.classes.tasks.Depositor']);
+
+		$this->upgrade();
+	}
+
+	/**
+	 * Upgrade and fixes
+	 */
+	protected function upgrade() {
+		// Before the version 2.0.4.3, it's needed to check for a missing "export_deposit_error" field
+		if (!Capsule::schema()->hasColumn('pln_deposits', 'export_deposit_error')) {
+			Capsule::schema()->table('pln_deposits', function (Blueprint $table) {
+				$table->string('export_deposit_error', 1000)->nullable();
+			});
+		}
+
+		// Changes introduced after version 2.0.4.3
+		if (!Capsule::schema()->hasColumn('pln_deposits', 'date_preserved')) {
+			Capsule::schema()->table('pln_deposits', function (Blueprint $table) {
+				$table->datetime('date_preserved')->nullable();
+				$table->string('staging_state')->nullable();
+				$table->string('lockss_state')->nullable();
+			});
+			// Reset status
+			Capsule::table('pln_deposits')->update(['status', null]);
```

Comment:
After some observations (reports from forum users + inspection of client installations + code review), I saw that the current state is not reliable, so here I'm resetting all states, they will be refilled later based on the status from the staging server.